test/externalTests/exttest/common.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2023 solidity contributors.
+# ------------------------------------------------------------------------------
+
+import os
+import sys
+import mimetypes
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+import subprocess
+from shutil import which, copyfile, copytree, rmtree
+from argparse import ArgumentParser
+
+from dataclasses import dataclass, field
+from typing import Dict, Optional, Tuple
+
+import re
+from abc import ABCMeta, abstractmethod
+
+# Our scripts/ is not a proper Python package so we need to modify PYTHONPATH to import from it
+# pragma pylint: disable=import-error,wrong-import-position
+SCRIPTS_DIR = Path(__file__).parents[3] / "scripts"
```

Comment:
Right. I changed it to this:
```python
PROJECT_ROOT = Path(__file__).parents[3]
sys.path.insert(0, f"{PROJECT_ROOT}/scripts")
```
Let me know what you think.

Reply:
I was thinking more of getting the root of the path in some other way, which would be more reliable. But I guess there's no other way than what you have done, since the script cannot know the absolute path of `PROJECT_ROOT`, only the relative one. This seems fine.